broadinstitute/single_cell_portal_core
# UI updates and Upload Feedback link (SCP-4897, SCP-5018, SCP-5017)

Fixes addressed in this PR:
- The loading spinner is more appropriately sized 
- On the Clustering tab you can now add multiple clustering forms
- Indications in both the Expression and Clustering steps and also the Nav Panel that you will be uploading the AnnData file at the end in the AnnData tab
- Add a button to toggle between AnnData and Classic mode if the user has not yet uploaded any file 

I plan to merge this after Jon's PR merges so will address conflicts here as needed.

To Test: 

NOTE - the changes here are almost entirely visual don't bother actually trying to upload here

- Pull this branch
- Start up local and navigate to a study that has no files yet
- When you click to the upload page observe the spinner 
- When you land on the option page observe the link to the survey
- Choose AnnData and see in the Navpanel at the bottom the link to the survey
- Hover over the "?" to read the text about uploading a file at the end
- Check out the Clustering tab and the Expression tab to see the text about uploading a file at the end
- In the Clustering tab observe that there is now a button "Add clustering" allowing for adding another form.



Improved spinner:

https://user-images.githubusercontent.com/54322292/225704711-8ec26e6f-979f-4747-a73c-0dddd28140e7.mov

On the option page the link to the survey
![Screen Shot 2023-03-10 at 12 23 48 PM](https://user-images.githubusercontent.com/54322292/225704742-6c6b8fef-da9b-4671-b769-52945ed93a0d.png)

The explanations in Clustering and Expression Matrices tabs that you upload at the end
![Screenshot 2023-03-13 at 1 54 43 PM](https://user-images.githubusercontent.com/54322292/225704871-236c0563-6df6-4989-bc84-db5b0d9a1aeb.png)
![Screenshot 2023-03-13 at 1 54 49 PM](https://user-images.githubusercontent.com/54322292/225704873-2abd6263-6ceb-4d08-83ed-2f930e281709.png)

Explanation about where to upload in the NavPAnel
![Screenshot 2023-03-13 at 2 14 31 PM](https://user-images.githubusercontent.com/54322292/225705001-95938dcc-a00c-451c-8577-a947e4fd4acf.png)
